### ISSUE TITLE
MS_EXPOSE_REP triggered for avro's BinaryMessageDecoder

### ISSUE THREAD

**Issue by hejcz** [spotbugs/spotbugs#1669]
I believe this is false positive. Why is `BinaryMessageDecoder` treated as an array?

```
MS: Public static method may expose internal representation by returning array (MS_EXPOSE_REP)

A public static method returns a reference to an array that is part of the static state of the class. Any code that calls this method can freely modify the underlying array. One fix is to return a copy of the array.
```

```
[INFO] --- spotbugs-maven-plugin:4.3.0:check (default) @ spotbugs-bug ---
[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[ERROR] Medium: Public static io.github.hejcz.TestClass.getDecoder() may expose internal representation by returning TestClass.DECODER [io.github.hejcz.TestClass] At TestClass.java:[line 16] MS_EXPOSE_REP
```

```java
package io.github.hejcz;

import org.apache.avro.message.BinaryMessageDecoder;
import org.apache.avro.specific.SpecificData;

public class TestClass {

    private static SpecificData MODEL$ = new SpecificData();

    public static final org.apache.avro.Schema SCHEMA$ = new org.apache.avro.Schema.Parser().parse("");

    private static final BinaryMessageDecoder<TestClass> DECODER =
            new BinaryMessageDecoder<TestClass>(MODEL$, SCHEMA$);

    public static BinaryMessageDecoder<TestClass> getDecoder() {
        return DECODER; // spotbugs detects bug in this line
    }

}
```

```xml
<plugin>
    <groupId>com.github.spotbugs</groupId>
    <artifactId>spotbugs-maven-plugin</artifactId>
    <version>4.3.0</version>
    <configuration>
        <failOnError>true</failOnError>
        <includeTests>false</includeTests>
    </configuration>
    <executions>
        <execution>
            <phase>compile</phase>
            <goals>
                <goal>check</goal>
            </goals>
        </execution>
    </executions>
</plugin>
```

**Comment by hejcz** [comment]
I tried to create an example but the test seems to pass. Maybe there is some problem with a plugin or the problem is fixed on master. 

An attempt to reproduce the problem: https://github.com/hejcz/spotbugs/commit/1be7ac048398b71bfb5cad96d06a5a6ebea6332b 
Project that actually fails: https://github.com/hejcz/spotbugs-avro-fp

**Comment by kuzmeo** [comment]
Hello!

Seems like issue is in com.github.spotbugsplugin version 4.3 and newer. I have the same issue when set report level to "medium"
[SpotbugsFalsePositive.zip](https://github.com/spotbugs/spotbugs/files/7561895/SpotbugsFalsePositive.zip)
.
I've attacheed sample, where issue can be easily reproduced. Just run gradle task spotbugsMain there. 
Class **SpotBugsFailsHere** there returns private static **final** field in public static method, but Spotbugs reports it as:
![image](https://user-images.githubusercontent.com/87804617/142395227-68408e8f-1b36-4c20-812d-6f8051668a99.png)

Regards,
Olga.

**Comment by gtoison** [comment]
`BinaryMessageDecoder ` is deemed mutable because it has the `addSchema()` method, see #1797 
The misleading message (only mentionning arrays) was fixed in #2535
